server/src/main/java/io/druid/client/cache/HybridCache.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.client.cache;
+
+import com.google.common.collect.Sets;
+import com.metamx.emitter.service.ServiceEmitter;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class HybridCache implements Cache
+{
+  private final Cache level1;
+  private final Cache level2;
+
+  private final AtomicLong hitCount = new AtomicLong(0);
+  private final AtomicLong missCount = new AtomicLong(0);
+
+  public HybridCache(Cache level1, Cache level2)
+  {
+    this.level1 = level1;
+    this.level2 = level2;
+  }
+
+  @Override
+  public byte[] get(NamedKey key)
+  {
+    byte[] res = level1.get(key);
+    if (res == null) {
+      res = level2.get(key);
```

Comment:
wondering if we should also warm up level1 cache when there is a miss in l1 cache and hit in l2 cache ?